### PR TITLE
Remove rhn and rhms/satellite columns

### PR DESCRIPTION
**Pull request by kbrock**
part of https://github.com/ManageIQ/manageiq/issues/5547

rhn went away in 2017 and we are no longer supporting satellite:

- Removing the columns that were used by the update manager and the subscription manager
- remove rhn and proxy authentication credentials
- remove setting for `update_repo_name`

### CODE REVIEW

**Comment by kbrock** [comment]
WIP: need to properly remove the rows associated with the subscription manager

File: db/migrate/20210722111142_remove_rhsm_columns.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  def up
+    change_table "miq_databases" do |t|
+      t.remove "registration_type"
+      t.remove "registration_organization"
+      t.remove "registration_server"
+      t.remove "registration_http_proxy_server"
+    end
+
+    Authentication.where(:authtype => %w[registration registration_http_proxy])
+                  .destroy_all
```
Comment (Fryguy) [inline db/migrate/20210722111142_remove_rhsm_columns.rb:15]: This should be wrapped in a `say_with_time`.  Also, I'm thinking this should be moved into a separate migration so that it can be tested independently.

File: db/migrate/20210722111142_remove_rhsm_columns.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    change_table "authentications" do |t|
+      t.remove "rhsm_sku"
+      t.remove "rhsm_pool_id"
+      t.remove "rhsm_server"
+      #t.remove "manager_ref"
```
Comment (Fryguy) [inline db/migrate/20210722111142_remove_rhsm_columns.rb:21]: This column should stay because it is for provider private key pairs. cc @agrare

File: db/migrate/20210722111142_remove_rhsm_columns.rb (hunk 1 above)
Comment (Fryguy) [reply db/migrate/20210722111142_remove_rhsm_columns.rb:15]: Also this needs a spec.

**Comment by Fryguy** [comment]
@agrare @bdunne please also review.

File: db/migrate/20210722111142_remove_rhsm_columns.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      t.remove "registration_http_proxy_server"
+    end
+
+    Authentication.where(:authtype => %w[registration registration_http_proxy])
+                  .destroy_all
+
+    change_table "authentications" do |t|
+      t.remove "rhsm_sku"
+      t.remove "rhsm_pool_id"
+      t.remove "rhsm_server"
```
Comment (bdunne) [inline db/migrate/20210722111142_remove_rhsm_columns.rb:20]: These are unrelated.  They were added in https://github.com/ManageIQ/manageiq/pull/7779/files#diff-49eb1e9f4c7c64a5250f45cd08b58cd7e0832c08f18c26e06082a972b8d183e7R29-R31

Comment (bdunne) [reply db/migrate/20210722111142_remove_rhsm_columns.rb:20]: Although it looks like the feature requiring them was removed in https://github.com/ManageIQ/manageiq/pull/18515/files#diff-031eafbc5af9f5092abc3e9c07627e482c156a37c0c9e5ff798089e0e61b0464L1

Comment (Fryguy) [reply db/migrate/20210722111142_remove_rhsm_columns.rb:20]: Oh that's right - this is related to ovirt authetnications. @agrare can you chime in here?

Comment (agrare) [reply db/migrate/20210722111142_remove_rhsm_columns.rb:20]: I think that's actually code for deploying openshift via automate (that's pretty cool actually)

Wasn't aware that existed before, but no it is definitely no longer used

File: db/migrate/20210722111142_remove_rhsm_columns.rb (hunk 2 above)
Comment (agrare) [reply db/migrate/20210722111142_remove_rhsm_columns.rb:21]: :+1: when we collect ssh keys from a provider we use this to store the provider native unique id, definitely still used

File: db/migrate/20210722111142_remove_rhsm_columns.rb (hunk 3 above)
Comment (Fryguy) [reply db/migrate/20210722111142_remove_rhsm_columns.rb:20]: If that's the case can _all_ of these go away? https://github.com/ManageIQ/manageiq/pull/7779/files#diff-49eb1e9f4c7c64a5250f45cd08b58cd7e0832c08f18c26e06082a972b8d183e7R6-R31

Comment (agrare) [reply db/migrate/20210722111142_remove_rhsm_columns.rb:20]: Maybe?  I didn't check that none of those are used anymore.

Either way probably should be pulled out of this PR since it isn't related to rhn/etc...

Comment (Fryguy) [reply db/migrate/20210722111142_remove_rhsm_columns.rb:20]: > Maybe? I didn't check that none of those are used anymore.

They all went in at the same time for the same feature, so if the feature is gone, I assumed they all should go

> Either way probably should be pulled out of this PR since it isn't related to rhn/etc...

Totally agree - doesn't belong in this PR.

Comment (kbrock) [reply db/migrate/20210722111142_remove_rhsm_columns.rb:13]: Can't remember who asked me to delete `RegistrationSystem`  "while I was at it"

So the goal is now to separate out the `UpdateSystem` from the `RegistrationSystem` and actually grow the `RegistrationSystem` (satellite) PR to delete more fields?

File: db/migrate/20210722111142_remove_rhsm_columns.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
+class RemoveRhsmColumns < ActiveRecord::Migration[6.0]
+  def up
+    change_table "miq_databases" do |t|
+      t.remove "registration_type"
+      t.remove "registration_organization"
+      t.remove "registration_server"
+      t.remove "registration_http_proxy_server"
```
Comment (bdunne) [inline db/migrate/20210722111142_remove_rhsm_columns.rb:7]: I think these columns belong in the migration above

File: db/migrate/20210722111142_remove_rhsm_columns.rb (hunk 3 above)
Comment (Fryguy) [reply db/migrate/20210722111142_remove_rhsm_columns.rb:13]: There are three completely different "satellite/registration" things in our product that are unrelated to each other.

1. Our appliance had the ability to update the RPMs on the appliance itself
2. The Satellite/Foreman provider is used for managing satellites
3. The OpenShift provider had a capability to be able to deploy new openshifts, and part of that effort was the need to store various credentials for that purpose (including credentials to an rhsm system)

This PR should focus only on number 1. The `rhsm_` columns in the authentications table are part of number 3.  I can't recall offhand which number `RegistrationSystem` belongs to, but I _thought_ it was number 1 ...perhaps @bdunne knows better.

Comment (bdunne) [reply db/migrate/20210722111142_remove_rhsm_columns.rb:13]: Yeah, `RegistrationSystem` is part of 1

**Comment by kbrock** [comment]
fixed copy paste and whitespace issues.

These new changes should be more focused on the columns that are just related to the registration system and update management

**Review by bdunne — APPROVED** [review]
👍🏻 LGTM

**Comment by bdunne** [comment]
Also, I opened https://github.com/ManageIQ/manageiq-style/pull/25 to take care of the rubocop comment

File: db/migrate/20210810191933_remove_update_repo_names_from_settings.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class RemoveUpdateRepoNamesFromSettings < ActiveRecord::Migration[6.0]
+  class SettingsChange < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
```
Comment (agrare) [inline db/migrate/20210810191933_remove_update_repo_names_from_settings.rb:3]: `SettingsChange` isn't an STI model so this isn't needed

**Review by agrare — APPROVED** [review]
